tests/tools/dumping/test_processes.py

```diff
@@ -0,0 +1,497 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Tests for the dumping of ProcessNode data to disk."""
+
+# For testing the dumping, one either needs to cd into the tmp_path, or pass the tmp_path as argument, otherwise, the
+# files are dumped into the src -> CWD from where the script is run.
+# However, when one passes tmp_dir as output_path, no automatic, default path is created
+
+from __future__ import annotations
+
+import io
+import shutil
+from pathlib import Path
+
+import pytest
+from aiida.common.links import LinkType
+from aiida.tools.dumping.processes import ProcessDumper
+
+# Non-AiiDA variables
+filename = 'file.txt'
+filecontent = 'a'
+inputs_relpath = Path('inputs')
+outputs_relpath = Path('outputs')
+node_inputs_relpath = Path('node_inputs')
+node_outputs_relpath = Path('node_outputs')
+default_dump_paths = [inputs_relpath, outputs_relpath, node_inputs_relpath, node_outputs_relpath]
+custom_dump_paths = [f'{path}_' for path in default_dump_paths]
+
+# Define variables used for constructing the nodes used to test the dumping
+singlefiledata_linklabel = 'singlefile'
+folderdata_linklabel = 'folderdata'
+folderdata_relpath = Path('relative_path')
+folderdata_test_path = folderdata_linklabel / folderdata_relpath
+arraydata_linklabel = 'arraydata'
+node_metadata_file = '.aiida_node_metadata.yaml'
+
+
+# Move this somewhere else?
+def clean_tmp_path(tmp_path: Path):
+    """Recursively delete files and directories in a path, e.g. a temporary path used by pytest."""
+
+    for item in tmp_path.iterdir():
+        if item.is_dir():
+            shutil.rmtree(item)
+        else:
+            item.unlink()
```

Comment:
Instead of this custom implementation, if you really need a clean empty path, either split the test in multiple smaller tests, so you don't need to clean manually. If you want to keep it as a single test, you can use [`tmp_path_factory`](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-factory-fixture) to create new directories using `tmp_path_factory.mktemp('some_dir')`.

Reply:
Done! Could even do without the `tmp_path_fatory` by updating the logic a bit in `test_validate_make_dump_path` :+1: